pkg/wshrpc/wshremote/wshremote_file.go

```
@@ -0,0 +1,810 @@
+	}
+	if append {
+		openFlags |= os.O_APPEND
+	}
+
+	file, err := os.OpenFile(path, openFlags, createMode)
+	if err != nil {
+		return fmt.Errorf("cannot open file %q: %w", path, err)
+	}
+	defer utilfn.GracefulClose(file, "RemoteWriteFileCommand", path)
+	if atOffset > 0 && !append {
+		n, err = file.WriteAt(dataBytes[:n], atOffset)
```

golangci-lint (2.5.0)

[error] ineffectual assignment to n

(ineffassign)